code/json.sls

```diff
@@ -30,172 +30,251 @@
 
 (library (json)
   (export json-write
-	  json-read)
-  
+          json-read)
+
   (import
-    (except (rnrs) define-record-type error)
-    (rnrs r5rs)
-    (rnrs mutable-pairs)
-    (only (srfi :1 lists) lset-union append-map fold)
-    (srfi :6 basic-string-ports)
-    (srfi :9 records)
-    (srfi :23 error)
-    (only (chezscheme) include)
-    (packrat))
+   (except (rnrs) define-record-type error)
+   (rnrs r5rs)
+   (rnrs mutable-pairs)
+   (only (srfi :1 lists) lset-union append-map fold)
+   (srfi :6 basic-string-ports)
+   (srfi :9 records)
+   (srfi :23 error)
+   (only (chezscheme) include)
+   (packrat))
 
+  ;; Provide the message 'pretty to get pretty printed output
+  ;; Otherwise pass any other symbol as the first argument
+  ;; and you will get regular json without formatting
```

Comment:
I need to update this comment, and then I should probably add an explanation of the interface

Reply:
Added explanation of the interface in a comment at the head of the function